packages/victory-core/src/victory-state/use-data.ts

```diff
@@ -0,0 +1,88 @@
+import { get, isNil, orderBy } from "lodash";
+import * as React from "react";
+import { Datum, DatumValue } from "../types/prop-types";
+import { isKeyValueObject } from "../victory-util/type-helpers";
+import { VictoryProviderProps } from "./types";
+
+type DataProps = Pick<
+  VictoryProviderProps,
+  "data" | "x" | "y" | "sortKey" | "sortOrder" | "samples"
+>;
+
+export interface FormattedDatum {
+  x: DatumValue;
+  y: DatumValue;
+  _x: number | Date;
+  _y: number | Date;
+  xName?: string;
+  yName?: string;
+  [key: string]: DatumValue;
+}
+
+function getValue(datum: Datum, key: string): DatumValue {
+  if (typeof datum === "object") {
+    return get(datum, key);
+  }
+  return datum;
+}
+
+function getNumericValue(value: DatumValue, fallback: number): number | Date {
+  if (typeof value === "number" || value instanceof Date) {
+    return value;
+  }
+  return fallback;
+}
+
+export function useData({
+  x: xAccessor = "x",
+  y: yAccessor = "y",
+  sortKey,
+  sortOrder = "ascending",
+  ...props
+}: DataProps) {
+  const data = React.useMemo<Datum[]>(() => {
+    if (props.data) {
+      return props.data;
+    }
+    return [];
+  }, [props.data]);
```

Comment:
`useMemo` not necessary
```suggestion
  const data = props.data || [];
```